### PR TITLE
[custom] Fix and Refactor react-error boundary

### LINKED ISSUE #223: [library]: create `react-error-boundary` library
#### Mockups

---

**[Contributor Guide](https://docs.meshery.io/project/contributing)**

- 🎨 Wireframes and [designs for Layer5 site](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR fixes #223 

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudhanshutech** [comment]
@nebula-aac should we also need to move this to root

**Comment by leecalcote** [comment]
@sudhanshutech there is new CSS for the MUIcheckbox's in Cloud that needs to be migrated to Sistent...

**Review by aabidsofi19 — APPROVED** [review]
As we dont need to make `react-error-boundary` external . we should remove the package  when we use it from sistent

**Comment by Yashsharma1911** [comment]
@sudhanshutech conflicts, can you update this error boundary component with support button you can copy component that we have in MeshMap

**Comment by sudhanshutech** [comment]
ok, are you referring for rjsf component? @Yashsharma1911

**Review by aabidsofi19 — APPROVED** [review]
Looks good

File: packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -1,88 +1,106 @@
...
+const ErrorMessage = styled(Typography)(() => ({
+  color: BLACK,
+  fontWeight: 'normal',
+  marginTop: '2px',
+  marginBottom: '2px',
+  fontSize: '1.15rem'
+}));
+
+const StyledLink = styled(Link)(() => ({
+  color: KEPPEL,
```
Comment (Yashsharma1911) [inline packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx:18]: Will this change according to theme?

File: packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -1,88 +1,106 @@
...
-*/
+export const withSuppressedErrorBoundary: React.FC<WithSuppressedErrorBoundaryProps> = ({
+  Component
+}: WithSuppressedErrorBoundaryProps): JSX.Element => {
+  return (
+    <ReactErrorBoundary FallbackComponent={Fallback} onError={reportError}>
+      <Component />
+    </ReactErrorBoundary>
+  );
+};
```
Comment (Yashsharma1911) [inline packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx:106]: Are we making sure that we can pass custom fallback component if we want

**Review by Yashsharma1911 — APPROVED** [review]

File: packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx (hunk 2 above)
Comment (aabidsofi19) [reply packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx:106]: suppressed should'nt have a fallback ( thats the purpose )

**Pull request merged**